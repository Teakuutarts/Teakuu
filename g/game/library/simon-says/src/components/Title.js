import React from "react";
import styled from "styled-components";
import { colors } from "../constants";

const Headline = styled.h1`
  color: #fff;
  font-size: 5em;
  text-align: center;
`;

const colorify = ({ colors, value }) => {
  const list = Object.keys(colors);
  const size = list.length - 1;
  const letters = [...Array(Math.floor(value.length / size))]
    .reduce((acc, n) => [...acc, ...list], [])
    .concat(list.slice(0, list.length % size))
    .map((x) => colors[x])
    .map((color, i) => (
      <span key={i} style={{ color }}>
        {value[i]}
      </span>
    ));

  return <span>{letters}</span>;
};

const Title = ({ value }) => <Headline>{colorify({ colors, value })}</Headline>;

export default Title;
