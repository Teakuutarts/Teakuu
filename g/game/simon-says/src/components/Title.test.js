import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Title from './Title';

it('renders correctly', () => {
  const wrapper = shallow(
    <Title value="foo bar" />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

