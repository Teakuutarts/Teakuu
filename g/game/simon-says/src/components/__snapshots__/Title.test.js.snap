exports[`test renders correctly 1`] = `
<styled.h1>
  <span>
    <span
      style={
        Object {
          "color": "#FF4136",
        }
      }>
      f
    </span>
    <span
      style={
        Object {
          "color": "#0074D9",
        }
      }>
      o
    </span>
    <span
      style={
        Object {
          "color": "#2ECC40",
        }
      }>
      o
    </span>
    <span
      style={
        Object {
          "color": "#FFDC00",
        }
      }>
       
    </span>
    <span
      style={
        Object {
          "color": "white",
        }
      }>
      b
    </span>
    <span
      style={
        Object {
          "color": "#1B1A1A",
        }
      }>
      a
    </span>
    <span
      style={
        Object {
          "color": "#FF4136",
        }
      }>
      r
    </span>
  </span>
</styled.h1>
`;
